### ISSUE TITLE
Remove hostNetwork: true from Node Daemonset

### ISSUE THREAD

**Issue by RyanStan** [kubernetes-sigs/aws-efs-csi-driver#1131]
To follow best practices, we should remove `hostNetwork: true` from the Node's Daemonset. However, removing it prevents IMDS access on clusters that have a limited network hop count. This prevents efs-utils from finding the region of a file system. Once efs-utils supports passing in the region as a command line option, we can modify this csi driver to pass the region, which we pull from the Kubernetes metadata (Node spec) to efs-utils.

Here is some more background and detail that I copied from a comment I made on another issue (https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/1111):

> The CSI Driver [uses the region it pulls from Kubernetes metadata to build a client to the EFS API](https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/master/pkg/cloud/cloud.go#L153). This is working as expected. However, the utility that the csi driver uses under the hood for performing mounts to EFS, [efs-utils](https://github.com/aws/efs-utils), requires [IMDS to find the region](https://github.com/aws/efs-utils/blob/master/src/mount_efs/__init__.py#L356-L388), which it then uses to construct the DNS name for the mount target. 
>
> The reason that I didn't run into this issue when initially trying to recreate it was because my efs-utils configuration file had been hardcoded with the correct region, so IMDS was not needed.
> 
> The immediate solution here is to add `hostNetwork=true` back into the Node Daemonset.

> And for the long term solution:
> I'd like to see us add a mount option in efs-utils to support configuring the region that way, instead of through the config file. There is already an open PR for this: https://github.com/aws/efs-utils/pull/171/files. Once that is merged, we can modify this driver to pass in the region it pulls from the Kubernetes Node spec as a mount option to efs-utils.
> 
> We will also need to update our testing infra to test against a IMDS disabled cluster. 

/feature

**Comment by alfredkrohmer** [comment]
I believe setting `hostNetwork: false` breaks all existing mounts and will keep breaking all mounts on every DaemonSet pod replacement, see my comment here: https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/1127#issuecomment-1734933606
